src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -148,37 +157,54 @@ public String produceDts(Compiler compiler) {
           isDefault = true;
         }
       }
-      emitNoSpace("declare namespace ");
-      emitNoSpace(INTERNAL_NAMESPACE);
-      if (!namespace.isEmpty()) {
-        emitNoSpace("." + namespace);
-      }
-      emitNoSpace(" {");
-      indent();
-      emitBreak();
-      if (symbol.getType() != null) {
-        walkScope(symbol, namespace, isDefault);
-      } else {
-        // JSCompiler treats "foo.x" as one variable name, so collect all provides that start with
-        // $provide + ".".
-        String prefix = provide + ".";
-        for (TypedVar other : topScope.getAllSymbols()) {
-          String otherName = other.getName();
-          if (otherName.startsWith(prefix) && other.getType() != null
-              && !other.getType().isFunctionPrototypeType()) {
-            walkScope(other, namespace, false);
-          }
-        }
-      }
-      unindent();
-      emit("}");
-      emitBreak();
+      currentNamespace = namespace;
```

Comment:
I agree, and fought with this for a while.
The problem is that we need to get a `relativeTo` all the way down in the visitor below, in `caseTemplatizedType`, and we need some significant refactoring to make a different instance of the visitor for each namespace, and pass this through the helper method calls in between.
I started down that path and backed out because of the scale of the change. But we could do that first as a prefactoring and then come back to this...